lib/pengine/unpack.c

```diff
@@ -908,80 +908,32 @@ unpack_resources(const xmlNode *xml_resources, pcmk_scheduler_t *scheduler)
     return TRUE;
 }
 
-/*!
- * \internal
- * \brief Unpack essential information from topology request XML
- *
- * \param[in]  xml     Request XML to search
- * \param[out] id      If not NULL, where to store level number
- *
- * \return Topology level XML from within \p xml, or NULL if not found
- * \note The caller is responsible for freeing \p *target and \p *desc if set.
- */
-static xmlNode *
-unpack_level_request(xmlNode *xml, int *id)
-{
-    int local_id = 0;
-
-    /* The level element can be the top element or lower. If top level, don't
-     * search by xpath, because it might give multiple hits if the XML is the
-     * CIB.
-     */
-    if ((xml != NULL) && !pcmk__xe_is(xml, PCMK_XE_FENCING_LEVEL)) {
-        xml = get_xpath_object("//" PCMK_XE_FENCING_LEVEL, xml, LOG_WARNING);
-    }
-
-    if (xml != NULL) {
-        crm_element_value_int(xml, PCMK_XA_INDEX, &local_id);
-    }
-
-    if (id != NULL) {
-        *id = local_id;
-    }
-
-    return xml;
-}
-
 /*!
  * \internal
  * \brief Parse configuration XML for fencing topology information
  *
  * \param[in]     xml_fencing_topology  Top of fencing topology configuration XML
  * \param[in,out] scheduler             Scheduler data
  *
- * \return TRUE
+ * \return void
  */
-gboolean
-unpack_fencing_topology(const xmlNode *xml_fencing_topology, pcmk_scheduler_t *scheduler)
+void
+pcmk__unpack_fencing_topology(const xmlNode *xml_fencing_topology, pcmk_scheduler_t *scheduler)
 {
     xmlNode *xml_obj = NULL;
     int id = 0;
-    xmlNode *level;
 
-    for (xml_obj = pcmk__xe_first_child(xml_fencing_topology, NULL, NULL, NULL);
-         xml_obj != NULL; xml_obj = pcmk__xe_next(xml_obj)) {
-
-        level = unpack_level_request(xml_obj, &id);
-        if (level == NULL) {
-            continue;
-        }
-
-        // Ensure an ID was given (even the client API adds an ID)
-        if (pcmk__str_empty(pcmk__xe_id(level))) {
-            pcmk__config_warn("Ignoring registration for topology level without ID");
```

Comment:
It's still worth checking the ID here (of xml_obj) since the fencer will reject a level without ID. The schema requires an ID but it's still worthwhile to check it in case validate-with="none".